src/alda/lisp/attributes.clj

```diff
@@ -74,3 +74,15 @@
   :aliases [:pan]
   :initial-val 0.5
   :transform percentage)
+
+(defn- parse-key-signature
+  "Transforms a key signature ((:c :sharp :sharp) (:d :flat)) into a letter
+   to accidentals map {:c (:sharp :sharp), :d (:flat)}."
+  [pitches]
+  (into {} (map (fn [pitch] {(first pitch) (rest pitch)}) pitches)))
+
+(defattribute key-signature
+   "The key in which the current instrument is playing."
+   :aliases [:key]
+   :initial-val {}
+   :transform (fn [val] (constantly (parse-key-signature (rest val)))))
```

Comment:
There are a couple things I don't understand here (possibly could be bugs?):

- In Clojure, [`constantly`](https://clojuredocs.org/clojure.core/constantly) is mostly used in the place of an anonymous function -- it returns an anonymous function that will always return a particular value. Because we want a function that returns a value, not a function that returns a function, I think either this shouldn't be wrapped in an anonymous `fn`, or the `constantly` shouldn't be there. Ideally this line should just be `:transform parse-key-signature`, and `parse-key-signature should be modified so that it ignores the first argument.

- I'm not following what the first (ignored) argument actually is :)  Why is `rest` necessary here?